### PR TITLE
feat: add proxyHost to reader/writer secrets

### PR DESCRIPTION
**Pull request by ahammond**
- feat: add proxyHost entry to user secrets
- ProxyEndpoint output should only exist if we are creating a proxy

Fixes #

### CODE REVIEW

File: src/aurora.ts [hunk 1]
```diff
@@ -279,36 +279,40 @@ export class Aurora extends Construct {
...
     });
+    this.secrets = secrets.map((s) => s.secret);
 
     if (!props.skipProxy) {
       this.proxy = new aws_rds.DatabaseProxy(this, 'Proxy', {
         dbProxyName: id.pascal,
         proxyTarget: aws_rds.ProxyTarget.fromCluster(this.cluster),
         //requireTLS: true, // If we're never allowing connections from outside the VPC, why bother?
-        secrets: this.secrets,
+        secrets: secrets.map((s) => s.secret),
```
Comment (chaocu) [inline src/aurora.ts:297]: Do we need to do this again after line 290 above?

**Review by Justinon — APPROVED** [review]
I ❤️ tests

File: src/aurora.ts (hunk 1 above)
Comment (Justinon) [inline src/aurora.ts:297]:
```suggestion
        secrets: this.secrets,
```

Comment (Justinon) [reply src/aurora.ts:297]: Yeah do we not have access to the `this` object in this line or am I missing something?

**Pull request merged**